### ISSUE TITLE
508-defect-1: Benefit summary and service verification letter accordion has critical heading, labelling, and responsive defects

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#45650]
## Point of contact

Josh Kim

## List of issues by severity
I've grouped these issues together as they're closely related and dependent on one another. Whatever solution we consider will likely need to take into account all of these issues together from a holistic perspective @skylerschain 

### Table isn't responsive (Critical)
The first table is not responsive and doesn't follow VA design system guidance
<img width="740" alt="responsivetables" src="https://user-images.githubusercontent.com/14154792/184412722-c1ae3049-6bb7-41fc-8bcb-8e289d2cca41.png">

### Headings aren't logically ordered (Major)
"Choose the information..." is a `h3` despite its parent being a `h3`; depending on intent, this may not be a heading at all.

<img width="1537" alt="headings" src="https://user-images.githubusercontent.com/14154792/184410904-effee184-6438-4033-ba4a-ba6a62bc5e82.png">

### No group labels (Major)
Several labelling issues exist with the way this accordion was designed:
- "Choose the information" isn't a `legend` within a `fieldset` nor a group label for the checkboxes below; screen reader users may critically miss important context on the purpose of those checkboxes as the key instruction "Choose the information" is not programmatically linked to the checkboxes.
- Similar to the rest of this page, the button is generically labelled

### Table checkbox pattern is too verbose for the aural experience (Major)

Due to the design, this pattern was coded in a manner where the aural experience is incredibly verbose. A screen reader user may hear the same labels repeated several times over along with row and column announcements.

https://user-images.githubusercontent.com/14154792/184919125-52872f6f-c229-4356-9aa5-1e622f0c52b4.MP4

### Download button needs review (Moderate)
The user is both generating the pdf and being navigated to a new page. We should clarify this in the language and styling of the component.

https://user-images.githubusercontent.com/14154792/184921694-3315ebc4-e781-447b-a942-357839887299.MP4

### Instructions rely on visual symbols (Moderate)
The first paragraph instructs users to expand boxes with "+"- a visual symbol that is not announced to screen reader users in the accordions. An immediate bandaid fix may be to align the aural experience with the instructions e.g., "To see an explanation of each letter, expand the _buttons_ below." However, this is likely a workaround to make up for an suboptimal design pattern.

https://user-images.githubusercontent.com/14154792/184918282-e7d465e0-0081-4fda-baed-65d5eaefcd53.MP4

### Responsive bugs (Moderate)
Buttons and tables cut off on mobile view. Alert is too narrow to read.
<img width="452" alt="responsivebug" src="https://user-images.githubusercontent.com/14154792/184414632-f53be4ff-7f27-4668-9197-2db6fe0a7fff.png">
<img width="407" alt="Screen Shot 2022-08-12 at 12 43 09 PM" src="https://user-images.githubusercontent.com/14154792/184414635-19a894e9-4770-40f7-9b73-872e46a6d973.png">

## Reproduction steps

1. [Log in as user 299 and get to step 2](https://staging.va.gov/records/download-va-letters/letters/letter-list)
2. Open the benefit summary accordion

## Proposed solution or next steps

TBD, this requires further exploration with design, engineering, and content. My immediate thoughts are:
- we should look into past research on why the designers choose the accordion and table pattern (for example, how complex can these tables be? e.g. for military service history); if no strong design reasons exist, we may want to consider redesigning this
- we should consider using plain checkboxes (instead of checkboxes within a table)
- depending on the length of this, it may deserve to live on a separate page 

## Type of issue

- [x] Screenreader
- [x] Keyboard
- [ ] Focus
- [x] Headings
- [ ] Color
- [x] Zoom
- [x] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by skylerschain** [comment]
We are hoping to be able to roll these fixes in with the Letters page updates as part of the Decision Letters MVF.

**Comment by jaredcunha** [comment]
Breaking this into separate tickets: 

- [Table isn't responsive (Critical)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50581)
- [Headings aren't logically ordered (Major)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/45944)
- [No group labels (Major)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50582)
- Table checkbox pattern is too verbose for the aural experience (Major) - ([Fixed](https://github.com/department-of-veterans-affairs/vets-website/pull/22288))
- [Download button needs review (Moderate)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/45998)
- [Instructions rely on visual symbols (Moderate)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/45999)
- [Responsive bugs (Moderate)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/45946)

**Comment by jacobworrell** [comment]
Closing this version of the ticket because it was broken into other items long ago and it's rendundant in our backlog.
